service/src/common/network_subgraph.rs

```diff
@@ -0,0 +1,192 @@
+            network_subgraph_url: Arc::new(
+                network_subgraph_url.expect("Could not query network subgraph"),
+            ),
+        }
+    }
+
+    pub async fn network_query_raw(
+        &self,
+        body: String,
+    ) -> Result<UnattestedQueryResult, reqwest::Error> {
+        let request = self
+            .client
+            .post(Url::clone(&self.network_subgraph_url))
+            .body(body.clone())
+            .header(header::CONTENT_TYPE, "application/json");
+
+        let response = request.send().await?;
+
+        // actually parse the JSON for the graphQL schema
+        let response_text = response.text().await?;
+        Ok(UnattestedQueryResult {
+            graphql_response: response_text,
+            attestable: false,
+        })
+    }
+
+    pub async fn network_query(
+        &self,
+        query: String,
+        variables: Option<Value>,
+    ) -> Result<UnattestedQueryResult, reqwest::Error> {
+        let body = GraphQLQuery { query, variables };
+
+        self.network_query_raw(
+            serde_json::to_string(&body).expect("serialize network GraphQL query"),
+        )
+        .await
+    }
+
+    pub async fn execute_network_free_query(
+        &self,
+        query: String,
+    ) -> Result<Response<UnattestedQueryResult>, QueryError> {
+        let response = self.network_query_raw(query).await?;
+
+        Ok(Response {
+            result: response,
+            status: 200,
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use wiremock::matchers::{method, path};
+    use wiremock::{Mock, MockServer, ResponseTemplate};
+
+    use super::*;
+
+    const GRAPH_NODE_STATUS_ENDPOINT: &str = "http://localhost:8000/";
+    const NETWORK_SUBGRAPH_ID: &str = "QmV614UpBCpuusv5MsismmPYu4KqLtdeNMKpiNrX56kw6u";
+    const NETWORK_SUBGRAPH_URL: &str =
+        "https://api.thegraph.com/subgraphs/name/graphprotocol/graph-network-goerli";
+
+    async fn mock_graph_node_server() -> MockServer {
+        let mock_server = MockServer::start().await;
+        let mock = Mock::given(method("POST"))
+            .and(path("/subgraphs/id/".to_string() + NETWORK_SUBGRAPH_ID))
+            .respond_with(ResponseTemplate::new(200).set_body_raw(
+                r#"
+                    {
+                        "data": {
+                            "graphNetwork": {
+                                "currentEpoch": 960
+                            }
+                        }
+                    }
+                "#,
+                "application/json",
+            ));
+        mock_server.register(mock).await;
+
+        mock_server
+    }
+
+    fn network_subgraph() -> NetworkSubgraph {
+        NetworkSubgraph::new(
+            Some(GRAPH_NODE_STATUS_ENDPOINT),
+            Some(NETWORK_SUBGRAPH_ID),
+            Some(NETWORK_SUBGRAPH_URL),
+        )
+    }
+
+    #[tokio::test]
+    #[ignore] // Run only if explicitly specified
+    async fn test_network_query_local() {
```

Comment:
This is not local, as it is querying thegraph.com. So perhaps this one should be `test_network_query` and the other one `test_network_query_mock`?